packager/media/codecs/h264_byte_to_unit_stream_converter.cc

```diff
@@ -24,6 +28,19 @@ H264ByteToUnitStreamConverter::H264ByteToUnitStreamConverter(
 
 H264ByteToUnitStreamConverter::~H264ByteToUnitStreamConverter() {}
 
+const H264Sps* RetrieveSps(std::vector<uint8_t> last_sps) {
+  const uint8_t* nalu_data = last_sps.data();
+  Nalu nalu;
+  if (!(nalu.Initialize(Nalu::kH264, nalu_data, last_sps.size())))
+    return nullptr;
+
+  int sps_id = 0;
+  H264Parser parser;
+  if (!(parser.ParseSps(nalu, &sps_id) == H264Parser::kOk))
+    return nullptr;
+  return parser.GetSps(sps_id);
+}
+
```

Comment:
1. Move the implementation inside the declaration in line 20. There is no need to have a declaration and an implementation later.
2. Rename it to ParseSpsFromBytes.
3. The code may crash as H264Sps is allocated on the stack which goes out of the scope when the function ParseSpsFromBytes exits. One option to workaround is to allocate H264Parser outside the function.
4. Declare |last_sps| as constant reference to avoid unnecessary copying.
5. Put it in an anonymous namespace, i.e. 

```c++
namespace {
const H264Sps* ParseSpsFromBytes(const std::vector<uint8_t>& sps, H264Parser* parser) {
  Nalu nalu;
  if (!nalu.Initialize(Nalu::kH264, sps.data(), sps.size()))
    return nullptr;

  int sps_id = 0;
  if (parser->ParseSps(nalu, &sps_id) != H264Parser::kOk)
    return nullptr;
  return parser->GetSps(sps_id);
} 
}  // namespace
```